fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java

```diff
@@ -198,11 +196,16 @@ private void mergeGroup(Group srcGroup, Group dstGroup) {
                 // set unused.
                 groupExpression.getGroup().removeGroupExpression(groupExpression);
                 groupExpression.setUnused(true);
+                GroupExpression existGroupExpression = groupExpressions.get(groupExpression);
```

Comment:
If the hash map only compute hash value once, or we cache the hash value, the merge process seems should be more simple.

Reply:
It is not more simple, if we cache the hash value,  at merge group phase, we also need to consider the group expresions which have same hashcode